File: ts/src/ordertaking/domain/unvalidatedOrder.ts
export class UnvalidatedOrder {
    constructor(
        readonly orderId: string,
        readonly customInfo: UnvalidatedCustomInfo,
        readonly shippingAddress: UnvalidatedAddress
    ) {}
}

class UnvalidatedCustomInfo {
    constructor(
        readonly firstName: string,
        readonly lastName: string,
        readonly emailAddress: string
    ) {
    }
}

class UnvalidatedAddress {
    constructor(
        readonly addressLine1: string,
        readonly addressLine2: string,
        readonly addressLine3: string,
        readonly addressLine4: string,
        readonly city: string,
        readonly zipCode: string
    ) {
    }
}

File: src/main/kotlin/ordertaking/domain/UnvalidatedOrder.kt
package com.rojojun.ordertaking.domain

data class UnvalidatedOrder(
    val orderId: String,
    val customInfo: UnvalidatedCustomInfo,
    val shippingAddress: UnvalidatedAddress,
)

data class UnvalidatedCustomInfo(
    val firstName: String,
    val lastName: String,
    val emailAddress: String,
)

data class UnvalidatedAddress(
    val addressLine1: String,
    val addressLine2: String,
    val addressLine3: String,
    val addressLine4: String,
    val city: String,
    val zipCode: String,
)